x-pack/platform/packages/private/kbn-infra-forge/src/lib/manage_template.ts

```diff
@@ -22,7 +22,7 @@ export async function installTemplate(
     await client.cluster
       .putComponentTemplate({
         name: component.name,
-        ...component.template,
+        ...(component.template as Omit<IndexTemplateDef, 'name'>),
```

Comment:
Why is the `as` necessary now? Is there any way to ensure the right type is there without that casting?

Reply:
I see this in a few places so I added a question about it to the main conversation of the PR: https://github.com/elastic/kibana/pull/208776#issuecomment-2666046970

Reply:
Without the casting, TS complains that the `name` assignment in line 24 is redundant (allegedly, `component.template` has a required `name` that overrides it because it's typed as `IndexTemplateDef`).

An alternative could be

```ts
.putComponentTemplate({
  ...component.template,
  name: component.name,
});
```

But I wasn't sure if we wanted to force `name` to be `component.name` or if we allowed the incoming `template` to override the name.